Add a blog post for Tenzir v4.9 (#3962)
tenzir/tenzir: web/blog/tenzir-v4.9/index.md

---
title: Tenzir v4.9
authors: [dominiklohmann]
date: 2024-02-21
tags: [context, bloom-filter, chart, dashboard]
comments: true
---

We're thrilled to announce the release of [Tenzir
v4.9](https://github.com/tenzir/tenzir/releases/tag/v4.9.0), enhancing the
Explorer further to empower you with the capability of rendering your data as a
chart.

![Tenzir v4.9](tenzir-v4.9.excalidraw.svg)

<!-- truncate -->

## Chart Operator

The new [`chart`](/next/operators/chart) operator transforms the way you
visualize your data on [app.tenzir.com](https://app.tenzir.com). It lets you
depict your events graphically instead of in table form.

Charting integrates seamlessly into your pipelines by simply adding the `chart`
operator. For instance, plotting a bar chart representing the frequency of

> occurences for each protocol in `suricata.flow` events can be as simple as this:

Reviewer: ```suggestion
occurrences for each protocol in `suricata.flow` events can be as simple as this:
```

> TODO: Add screenshots for all charts

Reviewer 2: Stub comment to address the TODO.

```
export
| where #schema == "suricata.flow"
| top proto
| chart bar --title "Protocols"
```

This line chart depicts the load average over 15 minutes for the past two days,
making use of the recently added `metrics` operator:

```
metrics
| where #schema == "tenzir.metrics.cpu"
| where #import_time > 2 days ago
| chart line -x timestamp -y loadavg_15m --title "Load Average (15 min)"
```

This area chart displays the total ingress across all pipelines for the past day
in MiB/s.

```
metrics
| where #schema == "tenzir.metrics.operator"
| where #import_time > 1 day ago

> | where source == true
> | where internal == false

Reviewer 2: ```suggestion
| where source == true && internal == false
```

YMMV, but throwing in a conjunction can't hurt.

| summarize egress=sum(output.approx_bytes), duration=sum(duration) by timestamp resolution 10 seconds
| sort timestamp
| python 'self.egress_rate = self.egress / self.duration.total_seconds() / 2**20'
| chart area -x timestamp -y egress_rate --title "Total Ingress MiB/s"
```

This pie chart shows the distribution of events stored at the node by disk
usage:

```
show partitions
| summarize diskusage=sum(diskusage) by schema
| python 'self.diskusage = self.diskusage / 2**30'
| chart pie --title "Disk Usage (GiB)"
```

We're just getting started with charting! If you want to see further chart types

> added, have feedback on charting, or want to share examples of what your

Author: ```suggestion
added, have feedback on charting, or want to share examples of your
```

visualizations with the chart operator, we would love to [hear from
you](/discord).

> :::info Coming Soon: Dashboards
> The `chart` operator is a first step towards having dashboards directly in
> Tenzir. Any result that you see in the Explorer you will soon be able to pin and
> freely arrange on customizable dashboard.
> :::

Reviewer: ```suggestion
:::info Coming Soon: Dashboards
The `chart` operator is a first step towards having dashboards directly in
Tenzir. Any result that you see in the Explorer you will soon be able to pin and
freely arrange on a customizable dashboard.
:::
```

| enrich indicators --field dns.rrname
```

> The `enrich` operator gained a new `--filter` option to remove events it could

Reviewer: Above this (can't directly comment on it because it's not part of this PR):
~~~
If you have massive amounts of indicators or a large amount of things you would
like>>>t<<< to contextualize, this feature is for you.
~~~

not enrich. Use the new option to remove anything that is not included in the
Bloom filter:

```
export --live
| where #schema == "suricata.dns"
| enrich indicators --field dns.rrname --filter
```

## Housekeeping

Other noteworthy changes and improvements:
- `tenzir.db-directory` is now `tenzir.state-directory`. The old option remains
  functional, but will be phased out in an upcoming release.
- On the command-line, Tenzir now respects [`NO_COLOR`](https://no-color.org)

>   when printing diagnostics.

Reviewer: And we do not print colors if we output device is not a terminal, making it suitable for non-interactive use, which is IMHO the more important part of the change. But you can leave as-is, I don't really mind.

Author: Added a sentence.

> - RFC 5424-style Syslog parsing no longer omits structured data fields.

Reviewer: Double negation.
```suggestion
- RFC 5424-style Syslog parsing now emits a record with the structured data fields.
```
